model-optimizer/mo/graph/perm_inputs.py

```diff
@@ -132,14 +132,21 @@ def shape(op_node: Node, port_info: str, input_port: int):
     assert permutation_data_node.has_and_set('permutation'), 'Data node "{}" does not have permutation for node {}, ' \
                                                              'port_info "{}".'.format(permutation_data_node.id,
                                                                                       op_node.id, port_info)
-    permutation = permutation_data_node.permutation
-    if len(permutation.perm) == 0:
+    permute_idx_for_gather = permutation_data_node.permutation.perm
+    if len(permute_idx_for_gather) == 0:
         return
 
+    # Kludge, 'it ain't much honest, but it works' since StridedSlice need to be permuted even inputs
+    # have rank lesser than 4, e.g. input_shape = (1, 10, 10) input[..., newaxis]
+    if op_node.type == 'StridedSlice':
+        from mo.ops.op import PermuteAttrs
+        slice_rank = op_node.in_port(input_port).data.get_shape()[0]
```

Comment:
this is not a rank. This is the first dimension of the shape.

`slice_rank = len(op_node.in_port(input_port).data.get_shape())`

Could you explain it?

Reply:
assume we have 4d input tensor and a slice `inp[0:1, 0:3]`, in that case slice rank will be 2. With the permutations we specified `op_node.in_port(input_port)` are inputs of `begin, end and strides` not for the input tensor! So
>op_node.in_port(input_port).data.get_shape()

will be [2] and

>len(op_node.in_port(input_port).data.get_shape()) 

will be 1. Therefore to get the rank of a slice we need `op_node.in_port(input_port).data.get_shape()[0]`

Reply:
These lines are not present any more. This logic have been moved to `check_shape(port_to_check)` function.

About necessity to permute even if input has rank <= 4 but slice_rank (first dimension of input) is >= 4 discussed personally and have written in comments here
1. `model-optimizer/mo/graph/perm_inputs.py:246`
2. `model-optimizer/mo/graph/perm_inputs.py:131`
3. `model-optimizer/extensions/middle/StridedSliceNormalizer.py:33`

Reply:
These lines are not present any more. This logic have been moved to `check_shape(port_to_check)` function.

About necessity to permute even if input has rank <= 4 but slice_rank (first dimension of input) is >= 4 discussed personally and have written in comments here
1. https://github.com/openvinotoolkit/openvino/blob/73046d7da4fad44ab682509cca69d370e7c79eda/model-optimizer/mo/graph/perm_inputs.py#L246
2. https://github.com/openvinotoolkit/openvino/blob/73046d7da4fad44ab682509cca69d370e7c79eda/model-optimizer/mo/graph/perm_inputs.py#L131
3. https://github.com/openvinotoolkit/openvino/blob/73046d7da4fad44ab682509cca69d370e7c79eda/model-optimizer/extensions/middle/StridedSliceNormalizer.py#L33